ToolBox/Polygon/PolygonDataQueueHandler.cs

```diff
@@ -507,105 +508,235 @@ private IEnumerable<Tick> GetCryptoTradeTicks(HistoryRequest request)
-                            yield break;
+                            // Iterate through each tick and process
+                            foreach (var objTick in objTicks)
+                            {
+                                // Parse the row
+                                var row = objTick.ToObject<EquityQuoteTickResponse>();
+
+                                // Converts the results tick time to the Exchange time zone for usage in the Tick Writer LeanData.cs
+                                var utcTime = Time.UnixNanosecondTimeStampToDateTime(row.SipTimestamp);
+                                var time = GetTickTime(request.Symbol, utcTime);
+
+                                // Get the timestamp of the last tick in the results and use for pagination of the API with offset
+                                if (objTicks.Count() == _apiResultsLimit)
+                                {
+                                    offsetStart = objTicks.Last.ToObject<EquityQuoteTickResponse>().SipTimestamp;
+                                }
+                                else
+                                {
+                                    offsetStart = offsetEnd;   // Finish
+                                }
+
+                                // TODO:   This logic should be enhanced by using the corrections to flag older ticks and better quote filtering
+                                // Look for quotes.   In this case, it uses the Polygon Quote conditions mapping for conditions that are suspicious and will be filtered
+                                bool _suspiciousFlag = false;
+
+                                // Set flag if there are any trade correction indicators    //TODO:   fix these quote and trade conditions logic ... not right !  Will bad quotes have null conditions ?
+                                if ((row.Conditions != null))
+                                {
+                                    _suspiciousFlag = (PolygonQuoteMappings.isSuspiciousQuote(row.Conditions[0]));
+                                }
+
+                                string quoteConditions = "";
+                                if (row.Conditions != null)
+                                {
+                                    quoteConditions = string.Join("", row.Conditions.Select(p => String.Format(System.Globalization.CultureInfo.GetCultureInfo("en-US"), "{0:X2}", p)));
+                                }
+                                // Adds the exchange the Tick.   Prefers the bid exchange and assumes mutually exclusive bid/ask exchange IDs on quote
+                                string exchange = "";
+                                if (row.BidExchange != null)
+                                {
+                                    exchange = row.BidExchange.ToString(System.Globalization.CultureInfo.GetCultureInfo("en-US"));
+                                }
+                                else if (row.AskExchange != null)
```

Comment:
`row.BidExchange` & `row.AskExchange` can't ever be null they are defined as an int

Reply:
@Martin-Molinero The Polygon API is NULL for the exchange since it separates the bid quote from the ask quote and can be different exchanges.   My understanding is that the LEAN data model combines these two into one field called Exchange

Reply:
If the Exchanges can be NULL in the response they should be nullable fields in the `EquityQuoteTickResponse` class, else they will keep their default value of `0`

Reply:
thanks, will fix.

Reply:
I spent some time thinking about this, and I believe that this code is the correct behavior.   The JSON parser will set the value to null if there is not a JSON field returned from the API.  The API doesn't return a value for that field.  If I am confused about the behavior, please let me know.

Reply:
>  The JSON parser will set the value to null if there is not a JSON field returned from the API

The parse will set the field to it's default value, for `int` it's 0, for it to be `null` the property should be `public int?` nullable see https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/nullable-value-types

Reply:
I will try that and see what happens.   Thanks.